R/fit_model.R

```diff
@@ -0,0 +1,179 @@
+#' Fit an EpiNow2 model
+#'
+#' @param data, in the format returned by [CFAEpiNow2Pipeline::read_data()]
+#' @param parameters As returned from
+#'   [CFAEpiNow2Pipeline::read_disease_parameters()]
+#' @param seed The random seed, used for both initialization by EpiNow2 in R and
+#'   sampling in Stan
+#' @param horizon The number of days, as an integer, to forecast
+#' @param priors A list of lists. The first level should contain the key `rt`
+#'   with elements `mean` and `sd` and the key `gp` with element `alpha_sd`.
+#' @param sampler_opts A list. The Stan sampler options to be passed through
+#'   EpiNow2. It has required keys: `cores`, `chains`, `iter_warmup`,
+#'   `iter_sampling`, `max_treedepth`, and `adapt_delta`.
+#'
+#' @return A fitted model object of class `epinow` or, if model fitting fails,
+#'   an NA is returned with a warning
+#' @export
+fit_model <- function(
+    data,
+    parameters,
+    seed,
+    horizon,
+    priors,
```

Comment:
Could add default priors here? `list("rt" = ...` If nothing else it'd make it more obvious about how to specify the priors having an example

Reply:
Or alternatively, perform a `setdiff()` on the names in the input list, against the necessary set of names?

If the user doesn't add, e.g. `priors[["rt"]][["mean"]]` to the list, wouldn't it return `NULL`? What would happen then? Would `EpiNow2::rt_opts()` return an error, or just silently continue on with some default value?

Reply:
I can, but I purposely didn't.

My thought was that we don't plan to use this function (or package) interactively. We're going to pass a task config with all parameters and won't depend on default arguments. If we misspecify something, we want an error not a silent reliance on a default argument with a (potentially) different value.

> If the user doesn't add, e.g. priors[["rt"]][["mean"]] to the list, wouldn't it return NULL? What would happen then? Would EpiNow2::rt_opts() return an error, or just silently continue on with some default value?

![image](https://github.com/user-attachments/assets/5444847e-9c9e-427d-b99a-22e45c649a81)


I think this error is the desired behavior. We don't _want_ the error, but with a config-driven rather than interactive workflow I think it's safer not to specify defaults.




But

Reply:
Yea, glad to see it'll error out if it doesn't have all the fields it needs.

Reply:
Arguably we could be making sure that the `priors` object is valid without using this function to catch it, but yes this justification makes sense to me